logs/field.go

```diff
@@ -0,0 +1,260 @@
+	return time.Nanosecond * time.Duration(f.vint)
+}
+
+// Strings is a value getter for fields with StringsType type
+func (f Field) Strings() []string {
+	if f.vany == nil {
+		return nil
+	}
+	return f.vany.([]string)
+}
+
+// Error is a value getter for fields with ErrorType type
+func (f Field) Error() error {
+	if f.vany == nil {
+		return nil
+	}
+	return f.vany.(error)
+}
+
+// Any is a value getter for fields with AnyType type
+func (f Field) Any() interface{} {
+	return f.vany
+}
+
+// Stringer is a value getteer for fields with StringerType type
+func (f Field) Stringer() fmt.Stringer {
+	if f.vany == nil {
+		return nil
+	}
+	return f.vany.(fmt.Stringer)
+}
+
+// Fallback returns default string representation of Field value.
+// It should be used by adapters that don't support f.Type directly.
+// Non-nil error indicates some fatal marshaling error and is most
+// likely a sign of invalid Field initialization.
+func (f Field) Fallback() (repr string, err error) {
+	defer func() {
+		if p := recover(); p != nil {
+			if e, ok := p.(error); ok {
+				err = e
+			} else {
+				err = fmt.Errorf("%v", p)
+			}
+		}
+	}()
+	switch f.ftype {
+	case IntType:
+		fallthrough
+	case Int64Type:
+		repr = strconv.FormatInt(f.vint, 10)
+	case StringType:
+		repr = f.vstr
+	case BoolType:
+		repr = strconv.FormatBool(f.Bool())
+	case DurationType:
+		repr = f.Duration().String()
+	case StringsType:
+		repr = fmt.Sprintf("%v", f.Strings())
+	case ErrorType:
+		repr = f.Error().Error()
+	case AnyType:
+		repr = fmt.Sprint(f.vany)
+	case StringerType:
+		repr = f.Stringer().String()
+	default:
+		err = fmt.Errorf("unknown FieldType %d", f.ftype)
+	}
+	return repr, err
+}
+
+// String constructs Field with StringType
+func String(key string, value string) Field {
+	return Field{
+		ftype: StringType,
+		key:   key,
+		vstr:  value,
+	}
+}
+
+// Int constructs Field with IntType
+func Int(key string, value int) Field {
+	return Field{
+		ftype: IntType,
+		key:   key,
+		vint:  int64(value),
+	}
+}
+
+func Int64(key string, value int64) Field {
+	return Field{
+		ftype: Int64Type,
+		key:   key,
+		vint:  value,
+	}
+}
+
+// Bool constructs Field with BoolType
+func Bool(key string, value bool) Field {
+	var vint int64
+	if value {
+		vint = 1
+	} else {
+		vint = 0
+	}
+	return Field{
+		ftype: BoolType,
+		key:   key,
+		vint:  vint,
+	}
+}
+
+// Duration constructs Field with DurationType
+func Duration(key string, value time.Duration) Field {
+	return Field{
+		ftype: DurationType,
+		key:   key,
+		vint:  value.Nanoseconds(),
+	}
+}
+
+// Strings constructs Field with StringsType
+func Strings(key string, value []string) Field {
+	return Field{
+		ftype: StringsType,
+		key:   key,
+		vany:  value,
+	}
+}
+
+// NamedError constructs Field with ErrorType. If value is nil,
+// resulting Field will be of AnyType instead of ErrorType.
+func NamedError(key string, value error) Field {
+	if value == nil {
+		return Any(key, nil)
```

Comment:
What about use ErrorType for nil error.

It allow skip nil errors in adapter and not show error fields for messages without error (as zap)